TCAT/Model/Direction.swift

```diff
@@ -83,70 +64,90 @@ class Direction: NSObject, NSCopying {
     var routeNumber: Int = 0
     
     /// An array of bus stop locations on the bus route, excluding the departure and arrival stop. Empty if `type != .depart`.
-    var stops: [LocationObject]
+    var stops: [LocationObject] = []
+    
+    /// Whether the user should stay on this direction's bus for an upcoming transfer.
+    var stayOnBusTransfer: Bool = false
+    
+    /// The unique identifier for the specific bus related to the direction.
+    var tripID: String = ""
     
     // MARK: Initalizers
 
     required init (
         type: DirectionType,
         name: String,
-        startLocation: CLLocationCoordinate2D,
-        endLocation: CLLocationCoordinate2D,
+        startLocation: LocationObject,
+        endLocation: LocationObject,
         startTime: Date,
         endTime: Date,
         path: [CLLocationCoordinate2D],
-        routeNumber: Int = 0,
-        stops: [LocationObject] = []
+        travelDistance: Double,
+        routeNumber: Int,
+        stops: [LocationObject],
+        stayOnBusTransfer: Bool,
+        tripID: String
     ) {
         self.type = type
         self.name = name
         self.startLocation = startLocation
         self.endLocation = endLocation
         self.startTime = startTime
-        self.path = path
         self.endTime = endTime
+        self.path = path
+        self.travelDistance = travelDistance
         self.routeNumber = routeNumber
         self.stops = stops
+        self.stayOnBusTransfer = stayOnBusTransfer
+        self.tripID = tripID
     }
 
     convenience init(name: String? = nil) {
 
-        let blankLocation = CLLocation()
+        let blankLocation = LocationObject.blank
         let blankTime = Date()
 
         self.init(
             type: .arrive,
             name: name ?? "",
-            startLocation: blankLocation.coordinate,
-            endLocation: blankLocation.coordinate,
+            startLocation: blankLocation,
+            endLocation: blankLocation,
             startTime: blankTime,
             endTime: blankTime,
-            path: []
+            path: [],
+            travelDistance: 0,
+            routeNumber: 0,
+            stops: [],
+            stayOnBusTransfer: false,
+            tripID: ""
         )
 
     }
 
     convenience init(from json: JSON) {
         
-        self.init()
+        // print("Direction JSON:", json)
         
-        type = {
-            switch json["type"].stringValue {
-            case "walk" : return .walk
-            case "depart" : return .depart
-            default : return .other
-            }
-        }()
+        self.init()
         
         name = json["name"].stringValue
+        type = json["type"].stringValue == "depart" ? .depart : .walk
```

Comment:
being kind of picky but lowercase the string value just in case

Reply:
It's in the REST doc that it should be that though, but I can

Reply:
yeah just in case. prob never happen but who knows